### ISSUE TITLE
Signal Handler not Implemented on Bash for Windows10

### ISSUE THREAD

**Issue by luastoned** [kernelsauce/turbo#302]
Hi,

Turbo installs fine on Bash for Windows10 (which is basically Ubuntu 14.04 LTS) but once you start a server there is an error and Ctrl+C isn't caught..
`[ioloop.lua] signalfd() in add_signal_handler() failed: Function not implemented`

Code here: https://github.com/kernelsauce/turbo/blob/master/turbo/ioloop.lua#L305-L311

**Comment by luastoned** [comment]
Here is the result from signalfd.lua (from examples)

	luajit signalfd.lua
	Press CTRL-C 3 times to exit
	[I 2017/01/10 17:06:44] [ioloop.lua] signalfd() in add_signal_handler() failed: Function not implemented

**Comment by luastoned** [comment]
Update: https://github.com/Microsoft/BashOnWindows/issues/129

**Comment by kernelsauce** [comment]
If its not supported then theres not much I can do to be honest...

**Comment by luastoned** [comment]
Yeah I realized that too, I wonder how Node and others handle it though.

**Comment by luastoned** [comment]
Any chance this could be sovled with [sigaction](http://man7.org/linux/man-pages/man2/sigaction.2.html)?
I tested this with Win10/WSL and it works.

**Comment by kernelsauce** [comment]
I suppose. I can investigate it further, but at the moment I'm pretty tied up with work related things so it won't be done straight away.

**Comment by luastoned** [comment]
I've played with sigaction and all early tests ended in
`PANIC: unprotected error in call to Lua API (bad callback)`

I hope you have more luck with it.
